lib/mss/content_protection.js

```diff
@@ -0,0 +1,338 @@
+    const PLAYREADY_RECORD_TYPES = ContentProtection.PLAYREADY_RECORD_TYPES;
+
+    const bytes = shaka.util.Uint8ArrayUtils.fromBase64(element.textContent);
+    const records = ContentProtection.parseMsPro_(bytes);
+    const record = records.filter((record) => {
+      return record.type === PLAYREADY_RECORD_TYPES.RIGHTS_MANAGEMENT;
+    })[0];
+
+    if (!record) {
+      return '';
+    }
+
+    const xml = shaka.util.StringUtils.fromUTF16(record.value, true);
+    const rootElement = shaka.util.XmlUtils.parseXmlString(xml, 'WRMHEADER');
+    if (!rootElement) {
+      return '';
+    }
+
+    return ContentProtection.getLaurl_(rootElement);
+  }
+
+  /**
+   * PlayReady Header format: https://goo.gl/dBzxNA
+   *
+   * @param {!Element} xml
+   * @return {?string}
+   * @private
+   */
+  static getKID_(xml) {
+    // KID element is optional and no more than one is
+    // allowed inside the DATA element.
+    for (const elem of xml.getElementsByTagName('DATA')) {
+      for (const child of elem.childNodes) {
+        if (child instanceof Element && child.tagName == 'KID') {
+          return child.textContent;
+        }
+      }
+    }
+
+    // Not found
+    return null;
+  }
+
+  /**
+   * Gets a PlayReady KID from a protection element
+   * containing a PlayReady Header Object
+   *
+   * @param {!Element} element
+   * @return {?string}
+   * @private
+   */
+  static getPlayReadyKID_(element) {
+    const ContentProtection = shaka.mss.ContentProtection;
+    const PLAYREADY_RECORD_TYPES = ContentProtection.PLAYREADY_RECORD_TYPES;
+
+    const bytes = shaka.util.Uint8ArrayUtils.fromBase64(element.textContent);
+    const records = ContentProtection.parseMsPro_(bytes);
+    const record = records.filter((record) => {
+      return record.type === PLAYREADY_RECORD_TYPES.RIGHTS_MANAGEMENT;
+    })[0];
+
+    if (!record) {
+      return null;
+    }
+
+    const xml = shaka.util.StringUtils.fromUTF16(record.value, true);
+    const rootElement = shaka.util.XmlUtils.parseXmlString(xml, 'WRMHEADER');
+    if (!rootElement) {
+      return null;
+    }
+
+    return ContentProtection.getKID_(rootElement);
+  }
+
+  /**
+   * Gets a initData from a protection element.
+   *
+   * @param {!Element} element
+   * @param {string} systemID
+   * @param {?string} keyId
+   * @return {?Array.<shaka.extern.InitDataOverride>}
+   * @private
+   */
+  static getInitDataFromPro_(element, systemID, keyId) {
+    const Uint8ArrayUtils = shaka.util.Uint8ArrayUtils;
+    const data = Uint8ArrayUtils.fromBase64(element.textContent);
+    const systemId = Uint8ArrayUtils.fromHex(systemID.replace(/-/g, ''));
+    const keyIds = new Set();
+    const psshVersion = 0;
+    const pssh =
+        shaka.util.Pssh.createPssh(data, systemId, keyIds, psshVersion);
+    return [
+      {
+        initData: pssh,
+        initDataType: 'cenc',
+        keyId: keyId,
+      },
+    ];
+  }
+
+  /**
+   * Creates DrmInfo objects from the given element.
```

Comment:
This description makes it sound like the method takes in a single element, but it takes in an array of elements instead.

Reply:
Added a better description